tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/cocosnet_evaluator.py

```diff
@@ -0,0 +1,532 @@
+        input_info = self.exec_network.input_info if has_info else self.exec_network.inputs
+        input_blob = next(iter(input_info))
+        with_prefix = input_blob.startswith('verification_network_')
+        if self.input_blob is None or with_prefix != self.with_prefix:
+            if self.input_blob is None:
+                output_blob = next(iter(self.exec_network.outputs))
+            else:
+                output_blob = (
+                    '_'.join(['verification_network', self.output_blob])
+                    if with_prefix else self.output_blob.split('verification_network_')[-1]
+                )
+            self.input_blob = input_blob
+            self.output_blob = output_blob
+            self.with_prefix = with_prefix
+            self.adapter.output_blob = output_blob
+
+    def fit_to_input(self, input_data):
+        return {self.input_blob: input_data}
+
+    def predict(self, identifiers, input_data):
+        predictions = self.exec_network.infer(self.fit_to_input(input_data))
+        result = self.adapter.process(predictions, identifiers, [{}])
+        return result
+
+
+class CocosnetModel:
+    def __init__(self, network_info, launcher, delayed_model_loading=False):
+        self.correspondence = CorrespondenceNetwork(network_info['correspondence_network'], launcher,
+                                                    delayed_model_loading)
+        self.generator = GeneratorNetwork(network_info['generative_network'], launcher, delayed_model_loading)
+
+    def release(self):
+        self.generator.release()
+        self.correspondence.release()
+
+    def predict(self, identifiers, inputs):
+        results = []
+        for current_input in inputs:
+            current_input = self.correspondence.fit_to_input(current_input)
+            corr_out = self.correspondence.predict(current_input)
+            gen_input = np.concatenate((corr_out[self.correspondence.key_of_warped_reference],
+                                        current_input[self.correspondence.key_of_input_semantics]), axis=1)
+            result = self.generator.predict(identifiers, gen_input)
+            results.append(*result)
+        return results
+
+
+class GanCheckModel(BaseModel):
+    def __init__(self, network_info, launcher, delayed_model_loading=False):
+        self.net_type = "verification_network"
+        self.additional_layers = network_info.get('additional_layers')
+        super().__init__(network_info, launcher)
+
+    def load_model(self, network_info, launcher, log=False):
+        model, weights = self.auto_model_search(network_info, self.net_type)
+        if weights:
+            self.network = launcher.read_network(model, weights)
+            for layer in self.additional_layers:
+                self.network.add_outputs(layer)
+            self.exec_network = launcher.ie_core.load_network(self.network, launcher.device)
+        else:
+            self.exec_network = launcher.ie_core.import_network(str(model))
```

Comment:
if you accept executable blob as model, then in this case you does not have network field and need to be careful with network calls. Also please check that it is accepted for all models. As I see, some of them directly works only with network